### ISSUE TITLE
Create a Pypi metadata parser

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#1260]
# Task Description

We need to implement a parser for the handling of various PyPi metadata files. Typically these files use `key:value` pairs (one per line).

For the moment, we should treat all the data fields as `String`-s.

This work will be needed for the PyPi layout provider's implementation and the `pypi-layout-provider` branch should be used for this.

## Sample Data

```
Metadata-Version: 1.0
Name: hello-strongbox-pip
Version: 1.0.0
Summary: Hello, Strongbox [pip]!
Home-page: https://github.com/strongbox/strongbox-examples
Author: Martin Todorov
Author-email: foo@bar.com
License: Apache 2.0
Description-Content-Type: UNKNOWN
Description: UNKNOWN
Platform: UNKNOWN
```

## Requirements

If possible, we'd like to be able to annotate the fields of DTO-s using some custom annotations.

Given the above sample data, we should be able to have a bean like this:

```
public class MyMetadata
{

    @PypiMetadataKey
    private String name;
    
    @PypiMetadataKey(name="Description-Content-Type")
    private String descriptionContentType;

    ....
}
```

# Task Relationships

This task:
* Is a sub-task of #807 

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

**Comment by carlspring** [comment]
@timur27,

As discussed, would you like to take this?

**Comment by timur27** [comment]
Hi @carlspring , 
yes, I would like to take this

**Comment by carlspring** [comment]
Implemented by @timur27 in #1288.
